a10_octavia/controller/worker/tasks/cert_tasks.py

```diff
@@ -63,143 +64,153 @@ class SSLKeyCreate(task.Task):
+                                                           cert=cert_data.cert_content,
+                                                           size=len(cert_data.cert_content),
+                                                           action=action,
+                                                           certificate_type=certificate_type)
+                    LOG.debug("SSL certificate updated successfully: %s", cert_data.cert_filename)
+            except Exception as e:
+                LOG.warning("Failed to update SSL Certificate: %s", str(e))
+            return cert_data
 
 
 class SSLKeyUpdate(task.Task):
     """Task to update an SSL Key"""
     @axapi_client_decorator
     def execute(self, cert_data, vthunder, action="import"):
-        try:
-            if self.axapi_client.file.ssl_key.exists(file=cert_data.key_filename):
-                self.axapi_client.file.ssl_key.update(file=cert_data.key_filename,
-                                                      cert=cert_data.key_content,
-                                                      size=len(cert_data.key_content),
-                                                      action=action)
-                LOG.debug("SSL Key updated successfully: %s", cert_data.key_filename)
-        except Exception as e:
-            LOG.warning("Failed to update SSL Key: %s", str(e))
+        if cert_data is not None:
+            try:
+                if self.axapi_client.file.ssl_key.exists(file=cert_data.key_filename):
+                    self.axapi_client.file.ssl_key.update(file=cert_data.key_filename,
+                                                          cert=cert_data.key_content,
+                                                          size=len(cert_data.key_content),
+                                                          action=action)
+                    LOG.debug("SSL Key updated successfully: %s", cert_data.key_filename)
+            except Exception as e:
+                LOG.warning("Failed to update SSL Key: %s", str(e))
 
 
 class ClientSSLTemplateUpdate(task.Task):
     """Task to update a client ssl template for a listener"""
     @axapi_client_decorator
     def execute(self, cert_data, vthunder):
-        try:
-            if self.axapi_client.slb.template.client_ssl.exists(name=cert_data.template_name):
-                self.axapi_client.slb.template.client_ssl.update(name=cert_data.template_name,
-                                                                 cert=cert_data.cert_filename,
-                                                                 key=cert_data.key_filename,
-                                                                 passphrase=cert_data.key_pass)
-                LOG.debug("Client SSL Template updated successfully: %s",
-                          cert_data.template_name)
-        except Exception as e:
-            LOG.warning("Failed to update Client SSL Template: %s", str(e))
+        if cert_data is not None:
```

Comment:
This shouldn't be needed. We've designed the flows so that cert_data will never be none if this task is executed.